ts/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -458,8 +458,8 @@ class LocalTrainingService implements TrainingService {
         await execMkdir(path.join(trialJobDetail.workingDirectory, '.nni'));
         await execNewFile(path.join(trialJobDetail.workingDirectory, '.nni', 'metrics'));
         const scriptName: string = getScriptName('run');
-        await fs.promises.writeFile(path.join(trialJobDetail.workingDirectory, scriptName),
-                                    runScriptContent.join(getNewLine()), { encoding: 'utf8', mode: 0o777 });
```

Comment:
In case you wonder, the default encoding of `writeFile()` is explicitly said to be utf8, regardless of system locale.